File: app/src/main/java/com/marked/pixsee/chat/custom/ImageHolder.java
package com.marked.pixsee.chat.custom;

import android.content.Context;
import android.net.Uri;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.facebook.common.util.UriUtil;
import com.facebook.drawee.view.SimpleDraweeView;
import com.marked.pixsee.R;
import com.marked.pixsee.chat.data.Message;
import com.marked.pixsee.chat.data.MessageConstants;

/**
 * Created by Tudor Pop on 04-Dec-15.
 */
public class ImageHolder extends RecyclerView.ViewHolder {
	private Context context;

	private SimpleDraweeView mImage;

	public ImageHolder(View itemView, Context context) {
		super(itemView);
		this.context = context;
		this.mImage = (SimpleDraweeView) itemView.findViewById(R.id.imageNetwork);
	}

	public void bindMessage(final Message message, final ChatAdapter.ChatInteraction chatInteraction) {
		itemView.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				chatInteraction.chatClicked(v, message, getAdapterPosition());
			}
		});
		String scheme = UriUtil.LOCAL_FILE_SCHEME;
		String path = message.getData().get(MessageConstants.DATA_BODY)
//				.replace(ServerConstants.SCHEME_HTTP, "")
//				.replace(ServerConstants.PORT, "/")
				;
		if (message.getMessageType() == MessageConstants.MessageType.YOU_IMAGE) {
			scheme = UriUtil.HTTP_SCHEME;
		}
		final Uri url = UriUtil.parseUriOrNull(path);

		mImage.setImageURI(url, context);
		mImage.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View v) {
				chatInteraction.imageClicked(v,url);
			}
		});
	}
}

File: app/src/main/java/com/marked/pixsee/views/adapter/viewholders/ImageHolder.kt
package com.marked.pixsee.views.adapter.viewholders;

import android.content.Context
import android.content.Intent
import android.net.Uri
import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.View
import com.facebook.binaryresource.FileBinaryResource
import com.facebook.drawee.view.SimpleDraweeView
import com.facebook.imagepipeline.cache.DefaultCacheKeyFactory
import com.facebook.imagepipeline.core.ImagePipelineFactory
import com.facebook.imagepipeline.request.ImageRequest
import com.marked.pixsee.R
import com.marked.pixsee.model.message.Message
import com.marked.pixsee.activity.chat.ImageFullscreenActivity
import org.jetbrains.anko.onClick

/**
 * Created by Tudor Pop on 04-Dec-15.
 */
class ImageHolder(itemView: View, val context: Context) : RecyclerView.ViewHolder(itemView){
	private val mImage = itemView.findViewById(R.id.imageNetwork) as SimpleDraweeView


    fun bindMessage(message: Message) {
//		val url = Uri.parse(message.data.get(MessageConstants.DATA_BODY));
		val url = Uri.parse("http://www.online-image-editor.com//styles/2014/images/example_image.png");
		mImage.setImageURI(url,context);
        mImage.onClick { largeImage(url) }
	}

    fun largeImage(uri: Uri){
        val imageRequest1= ImageRequest.fromUri(uri);
        val cacheKey1= DefaultCacheKeyFactory.getInstance().getEncodedCacheKey(imageRequest1);
        val resource = ImagePipelineFactory.getInstance().getMainDiskStorageCache().getResource(cacheKey1);
        val file=(resource as FileBinaryResource).getFile();
        Log.i("TAG", file.getAbsolutePath());

        val intent = Intent(context, ImageFullscreenActivity::class.java)
        intent.putExtra("URI",file.absolutePath)
        context.startActivity(intent)
    }


}
